.github/workflows/codeql-analysis.yml

```diff
@@ -7,34 +7,182 @@ on:
+    # At 01:37 every Thursday. Details in https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule
+    - cron: '37 1 * * 4'
 
 jobs:
+  create-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get languages from repo
+        id: set-matrix
+        uses: advanced-security/set-codeql-language-matrix@v1
+        with:
+          access-token: ${{ github.token }}
+          endpoint: ${{ github.event.repository.languages_url }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.languages }}
+
   analyze:
     name: Analyze
-    if: github.repository == 'photoview/photoview'
-    runs-on: ubuntu-20.04
-
-    # strategy:
-    #   fail-fast: false
-    #   matrix:
-        # Override automatic language detection by changing the below list
-        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
-        # language: ['go', 'javascript']
-        # Learn more...
-        # https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#overriding-automatic-language-detection
-
+    needs: create-matrix
+    if: ${{ needs.create-matrix.outputs.matrix != '[]' && github.repository == 'photoview/photoview' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ${{ ( matrix.language == 'go' && 'golang:1.23-bookworm' ) || ( matrix.language == 'javascript' && 'node:18' ) || 'null' }}
+      volumes:
+        - /opt/hostedtoolcache:/opt/hostedtoolcache
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ${{ fromJSON(needs.create-matrix.outputs.matrix) }}
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
+
+      - name: Install GO build dependencies
+        if: ${{ matrix.language == 'go' }}
+        working-directory: ./api
+        shell: bash
+        run: |
+          rm -rf /opt/hostedtoolcache/* # free up some space
+          git config --global --add safe.directory /__w/photoview/photoview
+          OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+          ARCH=$(uname -m)
+          case "$ARCH" in
+            x86_64) ARCH="amd64" ;;
+            aarch64) ARCH="arm64" ;;
+            armv6l) ARCH="arm/v6" ;;
+            armv7l) ARCH="arm/v7" ;;
+            *) echo "Unsupported architecture: $ARCH"; exit 1 ;;
+          esac
+          export TARGETPLATFORM="$OS/$ARCH"
+          echo "Platform is ${TARGETPLATFORM}"
+          export GOPATH="/go"
+          export PATH="${GOPATH}/bin:${PATH}"
+          export CGO_ENABLED=1
+          # Comment out the last line with 'apt-get install -y reflex sqlite3'
+          sed -i '$ s/^/#/' ../scripts/install_build_dependencies.sh
+          chmod +x ../scripts/*.sh
+          ../scripts/install_build_dependencies.sh
+          apt-get install -y file
+          echo "<<< Dependencies installed >>>"
+          . ../scripts/set_compiler_env.sh
```

Comment:
I think you should think about Code Analysis with amd64 only. Put this before `install_build_dependencies.sh` and only set `TARGETPLATFORM=linux/amd64`in the beginning. As it doesn't use `docker buildx` and `qemu`, it won't run with other arch.

Reply:
I set the platform in line 59.
I switched the order, but the result is the same - see the details in my message down the page